File: solutions/D. 더 게임 오브 데스.java
import java.util.Scanner;

public class Main {
    static class Game {
        int left;
        int right;
        Game(){
            left = -1;
            right = -1;
        }
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int k = s.nextInt();

        Game[] graph = new Game[n+1];
        for (int i = 1; i <= n; i++) {
            graph[i] = new Game();
            graph[i].left = s.nextInt();
            graph[i].right = s.nextInt();
        }

        int[] visited = new int[32];
        int remain = n;
        int now = k;

        while (remain != 1){
            int next = -1;
            if(visited[now] == 0){
                next = graph[now].left;
            }else{
                remain--;
                next = graph[now].right;
            }
            visited[now]++;
            now = next;
        }

        int result = -1;
        for(int i=1;i<=n;i++){
            if(visited[i] != 2){
                result = i;
                break;
            }
        }
        System.out.println(result);
    }
}

File: solutions/D. 더 게임 오브 데스.py
n,k = map(int,input().split())
graph = [[0,0]]

for _  in range(n):
    left,right = map(int,input().split())
    graph.append([left,right])

visited = [0]*32

remain,now = n,k
while remain != 1:
    next = -1
    if visited[now] == 0 :
        next = graph[now][0]
    else:
        next = graph[now][1]
        remain -= 1
    visited[now] += 1
    now = next

result = -1
for i in range(1,n+1):
    if visited[i] != 2:
        result = i
        break
print(result)